test/helpers/versions.spec.js

```diff
@@ -161,4 +240,19 @@ describe('helpers/versions', () => {
       versionsHelper.isPastLatest(qJson, '2.0.3').should.eql(true);
     });
   });
+  describe('.computeRangedVersion(currentVersion, newVersion)', () => {
+    it('should return new version as pinned, if current version is pinned', () => {
+      versionsHelper.computeRangedVersion('1.0.0', '2.0.0').should.eql('2.0.0');
+    });
+    it('should return `^` when single operand and `^` operator is used', () => {
+      versionsHelper.computeRangedVersion('^1.0.0', '2.0.0').should.eql('^2.0.0');
+    });
+    it('should return `~` when single operand and `~` operator is used', () => {
+      versionsHelper.computeRangedVersion('~1.0.0', '2.0.0').should.eql('~2.0.0');
+    });
+    it('complex range should append pinned version to existing range', () => {
+      versionsHelper.computeRangedVersion('~1.0.0 || ^2.0.0', '2.2.3')
+        .should.eql('~1.0.0 || ^2.0.0 || 2.2.3');
```

Comment:
I agree. This particular scenario would never happen. I've adjusted the test to use version ranges that make sense.